### ISSUE TITLE
Question about SAC Implementation: Design Rationale of `utd` Parameter

### ISSUE THREAD

**Issue by songyuc** [haosulab/ManiSkill#873]
Hi, guys!
In the SAC baseline implementation ([sac.py](https://github.com/haosulab/ManiSkill/blob/main/examples/baselines/sac/sac.py)), I noticed an interesting design choice regarding the `utd` (update-to-data ratio) parameter and would like to understand the rationale behind it.

**Current Implementation:**
```python
@dataclass
class Args:
    training_freq: int = 64  # training frequency (in steps)
    utd: float = 0.5  # update to data ratio
    # ...
    grad_steps_per_iteration: int = 0  # calculated as int(training_freq * utd)
```

**Observations:**
1. The gradient steps are coupled with data collection frequency through `utd` ratio
2. This creates an implicit dependency: changing `training_freq` affects the absolute number of gradient steps
3. Parameter interpretation becomes context-dependent (e.g., `utd=0.5` means different absolute steps when `training_freq` changes)

**Alternative Approach:**
Other implementations (e.g., [CleanRL](https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/sac_continuous_action.py)) use direct parameters:
```python
gradient_steps = 1  # absolute number of steps per iteration
train_frequency = 64  # data collection before updating
```

**Questions:**
1. What was the design rationale behind using the `utd` ratio instead of direct gradient step parameters?
2. Is there a performance consideration (e.g., GPU utilization) that makes this coupling beneficial?
3. Are there plans to decouple these parameters for more flexible hyperparameter tuning?
4. How does this design interact with the parallel environment architecture (`num_envs=16`)?

**References:**
- [Parameter calculation](https://github.com/haosulab/ManiSkill/tree/main/examples/baselines/sac/sac.py#L274-L276)
- [Training loop usage](https://github.com/haosulab/ManiSkill/tree/main/examples/baselines/sac/sac.py#L478)

Thank you for maintaining this excellent framework! I'm asking this to better understand the design decisions behind industrial-grade RL implementations.

**Comment by StoneT2000** [comment]
1. Many offline rl research and off policy rl research tend to measure some notion of utd, as increasing this directly corresponds with higher sample efficiency (more gradients for the same amount of data). It is more of a preference that our team liked to use following recent research. If you have a different preference you can modify the code to be like CleanRL directly
2. No performance changes, it is just differnet ways to control the same things
3. Probably not as it is unclear what is more preferred so we just provide an example and let users change as needed
4. See the code, UTD is based on the amount of data collected across all environments.
